src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts

```diff
@@ -0,0 +1,59 @@
+import type { Order2FulfillmentDetailsCompletedViewProps } from "Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/Order2FulfillmentDetailsCompletedView"
+import type { useCompleteFulfillmentDetailsData_order$key } from "__generated__/useCompleteFulfillmentDetailsData_order.graphql"
+import { graphql, useFragment } from "react-relay"
+
+/**
+ * Hook that returns props for the completed fulfillment details view if the step is complete.
+ * Returns null if fulfillment details are not complete.
+ * Fulfillment details is complete if the backend has saved minimal address data.
+ */
+export const useCompleteFulfillmentDetailsData = (
+  order: useCompleteFulfillmentDetailsData_order$key,
+): Order2FulfillmentDetailsCompletedViewProps | null => {
+  const orderData = useFragment(FRAGMENT, order)
+
+  const details = orderData.fulfillmentDetails ?? {}
+  const isComplete = ["addressLine1", "country"].every(field => {
+    return details[field as keyof typeof details] != null
+  })
+
+  if (!isComplete) {
+    return null
+  }
+
+  const isPickup = orderData.selectedFulfillmentOption?.type === "PICKUP"
+
+  return {
+    isPickup,
+    fulfillmentDetails: {
+      name: orderData.fulfillmentDetails?.name || null,
```

Comment:
can we use the details object on line 15 here as well?

Reply:
refactored a bit here - in short the hand-written prop makes this harder to reuse because the properties are `readonly` coming from relay. it's a papercut but a little cleaner with the change.